packages/fast-components-react-base/src/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import Foundation, { HandledProps } from "../foundation";
+import { ICheckboxHandledProps, ICheckboxManagedClasses, ICheckboxUnhandledProps } from "./checkbox.props";
+import { ICheckboxClassNameContract, IManagedClasses } from "@microsoft/fast-components-class-name-contracts";
+import { get } from "lodash-es";
+
+/* tslint:disable-next-line */
+class Checkbox extends Foundation<ICheckboxHandledProps & ICheckboxManagedClasses, ICheckboxUnhandledProps, {}> {
+    protected handledProps: HandledProps<ICheckboxHandledProps & IManagedClasses<ICheckboxClassNameContract>> = {
+        managedClasses: void 0,
+        text: void 0,
+        checked: void 0,
+        indeterminate: void 0,
+        onChange: void 0
+    };
+
+    /**
+     * Apply indeterminate state to items that are indeterminate.
+     * This method should be called after render because it relies on element references.
+     */
+    public applyIndeterminateState(): any {
+        const ref: React.ReactNode = this.getRef("input");
```

Comment:
React 16.3 just came out with a new ref API that I believe will be better suited for this use case: https://reactjs.org/blog/2018/03/29/react-v-16-3.html - I would recommend using this API instead of `getRef`/`setRef`